### PR TITLE
feature: new design for Connections setting pane

### LINKED ISSUE #6720: Redesign the "Connections" settings section to match the new designs
## Describe the solution you'd like

We have new designs and language for the Connections (formerly Adapters) part of the bot settings, and these should be implemented.

### PR DESCRIPTION
**Pull request by beyackle**
## Description

This remodels the "Connections" part of the setting pane to align with the new designs. The "Publish profiles" section above it will need to be moved later, and there were "Learn more" links in the design that will need to be added back once we have URLs to point them to.

## Task Item

closes #6720 

## Screenshots

image

### CODE REVIEW

File: Composer/packages/client/src/pages/botProject/adapters/AdapterSection.tsx [hunk 1]
```diff
@@ -39,12 +27,25 @@ const AdapterSection = ({ projectId, scrollToSectionId }: Props) => {
...
-      <CollapsableWrapper title={formatMessage('Adapters')} titleStyle={title}>
-        <div css={subtitle}>{formatMessage('Connect your bot to other messaging services.')}</div>
-        {renderSectionHeader(formatMessage('Azure Bot Service channels'), '(description of ABS channels)')}
-        <ABSChannels projectId={projectId} />
-        {renderSectionHeader(formatMessage('External service adapters'), '(description of external adapters)')}
-        <ExternalAdapterSettings projectId={projectId} />
+      <CollapsableWrapper title={formatMessage('Connections')} titleStyle={title}>
+        <Stack
+          tokens={{
+            padding: '6px 0 6px 0',
```
Comment (srinaath) [inline Composer/packages/client/src/pages/botProject/adapters/AdapterSection.tsx:33]: padding: '6px 0'

File: Composer/packages/client/src/pages/botProject/styles.ts [hunk 2]
```diff
@@ -6,18 +6,17 @@ import { NeutralColors, SharedColors } from '@uifabric/fluent-theme';
...
 `;
 
 export const subtitle = css`
   color: ${NeutralColors.gray130};
-  font-size: ${FontSizes.smallPlus};
-  & > h1 {
-    margin-top: 0;
-  }
+  font-size: ${FontSizes.medium};
+
```
Comment (srinaath) [inline Composer/packages/client/src/pages/botProject/styles.ts:18]: nit: can remove space here

File: Composer/packages/client/src/pages/botProject/adapters/ExternalAdapterSettings.tsx [hunk 3]
```diff
@@ -133,31 +101,11 @@ const ExternalAdapterSettings = (props: Props) => {
...
-                styles={{ root: { paddingTop: '10px', paddingBottom: '10px' } }}
-                onKeyDown={(e) => {
-                  if (e.key === 'Enter') {
-                    e.stopPropagation();
-                  }
-                }}
-              />
-            </TooltipHost>
+            <div css={tableRowItem(columnSizes[2])} role="cell">
+              <Link key={key} onClick={() => openModal(key, packageName)}>
```
Comment (srinaath) [inline Composer/packages/client/src/pages/botProject/adapters/ExternalAdapterSettings.tsx:105]: can we rename event handler as on{actionName}{eventName}. essentially, maybe onOpenModalClick?

File: Composer/packages/client/src/pages/botProject/adapters/ExternalAdapterSettings.tsx [hunk 4]
```diff
@@ -40,40 +36,29 @@ const ExternalAdapterSettings = (props: Props) => {
...
     } else {
-      setModalProps({ key, packageName, firstTime });
+      setModalProps({ key, packageName });
     }
   };
 
   if (schemaDefinitions == null) return null;
 
-  const externalServices = (schemas: (JSONSchema7 & { key: string; packageName?: string; firstTime?: boolean })[]) => (
+  const externalServices = (schemas: (JSONSchema7 & { key: string; packageName?: string })[]) => (
```
Comment (srinaath) [inline Composer/packages/client/src/pages/botProject/adapters/ExternalAdapterSettings.tsx:51]: There are few places this type is being used { key: string; packageName?: string }. Maybe create a type to be used across files.

In addition it looks like key always exist as a string and packageName could be undefined. There are places where it is indicated key and packageName could be optional. Could we make sure this type is consistent?

**Review by tonyanziano — APPROVED** [review]

**Comment by cwhitten** [comment]
@beyackle there is a conflict on this now. can you fix it up? thanks

**Review by tonyanziano — APPROVED** [review]

**Pull request merged**